appserver/orb/orb-connector/src/main/java/org/glassfish/enterprise/iiop/api/GlassFishORBHelper.java

```diff
@@ -112,55 +103,54 @@ public ORB getORB() {
         // Use a volatile double-checked locking idiom here so that we can publish
         // a partly-initialized ORB early, so that lazy init can come into getORB()
         // and allow an invocation to the transport to complete.
-        if (orb == null) {
-
-            synchronized (this) {
-                if (orb == null) {
-                    try {
-                        boolean isServer = processEnv.getProcessType().isServer();
-
-                        Properties props = new Properties();
-                        props.setProperty(GlassFishORBFactory.ENV_IS_SERVER_PROPERTY, Boolean.valueOf(isServer).toString());
+        if (orb != null || destroyed) {
```

Comment:
A small note on optimization ))

We can read volatile `orb` field into a local variable and work with it. This will reduce number of the volatile reads.

Reply:
As it is outside the sync block it could even change between usages. It is not much probable here, but it was still a mistake. I have extracted the initProtocolManager method with the ORB parameter too.